### ISSUE TITLE
Suggestion to add include directories to interface of target "fltk" in CMake

### ISSUE THREAD

**Issue by holyTyu** [fltk/fltk#207]
CMake has a useful feature for installed targets to have include directories specified through the target property INTERFACE_INCLUDE_DIRECTORIES ([CMake docs](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html)).

You can populate the property by this lines in main CMakeLists.txt (adapted the example from CMake docs):
```
target_include_directories( fltk INTERFACE
  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
  $<INSTALL_INTERFACE:include>
)
```

What this allows is to write CMakeLists.txt like this:
```
add_executable( app main.cpp )
target_link_libraries( app PRIVATE fltk )
```
Instead of:
```
add_executable( app main.cpp )
target_link_libraries( app PRIVATE fltk )
target_include_directories( app PRIVATE ${FLTK_INCLUDE_DIRS} )
```

It is a nice QoL improvement and kind of expected from imported targets to bring their headers with them.

**Comment by Albrecht-S** [comment]
Thank you very much for this proposal. Thanks also for the code example (very helpful!) and the link to the docs.

It's on my todo list to improve FLTK's CMake support with things like this. This will certainly be integrated in FLTK 1.4.0.

**Comment by Albrecht-S** [comment]
This issue has been fixed by commit fd5cd809356dc73d2ede5bb2f0db25098771cb8e. FLTK supports now "Modern CMake" which allows to link CMake "targets" and get everything needed from the target's definition.

Please confirm and close this issue. Otherwise this issue will be closed in a few days.
